### ISSUE TITLE
 System.Net.Http.Functional.Tests.SocketsHttpHandlerTest_Http2/GoAwayFrame_AbortAllPendingStreams_StreamFailWithExpectedException failing in CI

### ISSUE THREAD

**Issue by Jozkee** [dotnet/corefx#38761]
From #38566

https://mc.dot.net/#/user/dotnet-bot/pr~2Fdotnet~2Fcorefx~2Frefs~2Fpull~2F38566~2Fmerge/test~2Ffunctional~2Fcli~2Finnerloop~2F/20190619.80/workItem/System.Net.Http.Functional.Tests/analysis/xunit/System.Net.Http.Functional.Tests.SocketsHttpHandlerTest_Http2~2FGoAwayFrame_AbortAllPendingStreams_StreamFailWithExpectedException

Message:
```
System.Net.Http.HttpRequestException : An error occurred while sending the request.
---- System.ObjectDisposedException : Cannot access a disposed object.
Object name: 'CreditManager'.
```

Stack Trace:
```  
at System.Net.Http.Http2Connection.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken) in /_/src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs:line 1449
   at System.Net.Http.HttpConnectionPool.SendWithRetryAsync(HttpRequestMessage request, Boolean doRequestAuth, CancellationToken cancellationToken) in /_/src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs:line 539
   at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken) in /_/src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/RedirectHandler.cs:line 33
   at System.Net.Http.HttpClient.FinishSendAsyncBuffered(Task`1 sendTask, HttpRequestMessage request, CancellationTokenSource cts, Boolean disposeCts) in /_/src/System.Net.Http/src/System/Net/Http/HttpClient.cs:line 496
   at System.Net.Http.Functional.Tests.HttpClientHandlerTest_Http2.<>c__DisplayClass39_0.<<EstablishConnectionAndProcessOneRequestAsync>b__0>d.MoveNext() in /_/src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Http2.cs:line 739
--- End of stack trace from previous location where exception was thrown ---
   at System.Threading.Tasks.TaskTimeoutExtensions.WhenAllOrAnyFailed(Task[] tasks) in /_/src/Common/tests/System/Threading/Tasks/TaskTimeoutExtensions.cs:line 83
   at System.Threading.Tasks.TaskTimeoutExtensions.WhenAllOrAnyFailed(Task[] tasks) in /_/src/Common/tests/System/Threading/Tasks/TaskTimeoutExtensions.cs:line 111
   at System.Threading.Tasks.TaskTimeoutExtensions.WhenAllOrAnyFailed(Task[] tasks, Int32 millisecondsTimeout) in /_/src/Common/tests/System/Threading/Tasks/TaskTimeoutExtensions.cs:line 71
   at System.Net.Http.Functional.Tests.HttpClientHandlerTest_Http2.EstablishConnectionAndProcessOneRequestAsync(HttpClient client, Http2LoopbackServer server) in /_/src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Http2.cs:line 735
   at System.Net.Http.Functional.Tests.HttpClientHandlerTest_Http2.GoAwayFrame_AbortAllPendingStreams_StreamFailWithExpectedException() in /_/src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Http2.cs:line 889
--- End of stack trace from previous location where exception was thrown ---
----- Inner Stack Trace -----
   at System.Net.Http.CreditManager.RequestCreditAsync(Int32 amount, CancellationToken cancellationToken) in /_/src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/CreditManager.cs:line 50
   at System.Net.Http.Http2Connection.SendHeadersAsync(HttpRequestMessage request, CancellationToken cancellationToken, Boolean mustFlush) in /_/src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs:line 972
   at System.Threading.Tasks.ValueTask`1.get_Result() in /_/src/System.Private.CoreLib/shared/System/Threading/Tasks/ValueTask.cs:line 763
   at System.Net.Http.Http2Connection.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken) in /_/src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs:line 1370
```

**Comment by davidsh** [comment]
@wfurt

**Comment by wfurt** [comment]
this should be fixed by #38226

**Comment by krwq** [comment]
I haven't seen any intermittent failures here recently. Should we close this until we got another repro?
